client/gutenberg/extensions/publicize/connection.jsx

```diff
@@ -12,14 +12,27 @@
  */
 import { Component } from '@wordpress/element';
 import { Disabled, FormToggle, Notice, ExternalLink } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
 import { withSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
 import PublicizeServiceIcon from './service-icon';
+import getSiteFragment from 'gutenberg/extensions/presets/jetpack/editor-shared/get-site-fragment';
 
+/**
+ * Return a link to the Sharing page, whether it's on Calypso or WP Admin.
+ *
+ * @returns {string} Link to Sharing page.
+ */
+const getSharingLink = () =>
+	getSiteFragment()
+		? // If running in WP.com wp-admin or in Calypso, we redirect to Calypso sharing settings.
+		  `https://wordpress.com/sharing/${ getSiteFragment() }`
+		: // If running in WordPress.org wp-admin we redirect to Sharing settings in wp-admin.
+		  'options-general.php?page=sharing&publicize_popup=true';
```

Comment:
This relative URL seems pretty fragile.

Reply:
Indeed, we can build this with `addQueryArgs` from `@wordpress/url`

Reply:
I'm not completely up to date but wasn't this redirect going away and we'll always redirect to wp.com?

Reply:
Last thing I remember was that instead of redirecting to WP.com, we'd keep the user in the same page when managing publicize services - this could happen by building a new modal for that purpose, see #28387

Reply:
@tyxla that section has been removed and I think the plan is to keep it in Calypso only: https://github.com/Automattic/jetpack/pull/11314

The URL/page still exists, there's just a button that brings you to Calypso so it's an extra step we can avoid by bringing folks always to Calypso. Suppose that means we'll need to clarify/warn that in the copy at each place where we're doing that jump so that it doesn't come as a surprise.